main/src/library/pkger/Manifest.flix

```diff
@@ -0,0 +1,287 @@
+
+        // Project directories
+        let getBuildDir = (base) -> Path.add(getProjectDir(base), "build");
+        let getSourceDir = (base) -> Path.add(getProjectDir(base), "src");
+        let getTestDir = (base) -> Path.add(getProjectDir(base), "test");
+
+        // Top-level files
+        let getPackageFile = (base) -> Path.add(getProjectDir(base), "package.sn");
+        let getHistoryFile = (base) -> Path.add(getProjectDir(base), "HISTORY.md");
+        let getLicenseFile = (base) -> Path.add(getProjectDir(base), "LICENSE.md");
+        let getReadmeFile = (base) -> Path.add(getProjectDir(base), "README.md");
+
+        // Source files
+        let getMainSourceFile = (base) -> Path.add(getSourceDir(base), "Main.flix");
+        let getMainTestFile = (base) -> Path.add(getTestDir(base), "TestMain.flix");
+
+        let newFile = (content) -> (path) -> {
+            Path.writeLines(path, Array.toList(content))
+        };
+
+        let workingDir = Path.workingDirectory();
+
+        let files = Array.toList([getPackageFile, getHistoryFile, getLicenseFile, getReadmeFile, getMainSourceFile, getMainTestFile]);
+        let dirs = Array.toList([getBuildDir, getSourceDir, getTestDir]);
+
+        // Ensure that none of the paths the init process creates are currently present on the system
+        let pathExists = [files, dirs]
+            |> Array.toList
+            |> List.flatten
+            |> List.map(f -> {
+                    let dir = Result.map(f, workingDir);
+                    match Result.flatMap(Path.exists, dir) {
+                        case Ok(exists) => exists,
+                        case _ => true
+                    }
+                })
+            |> List.find(x -> x);
+
+        if (Option.isEmpty(pathExists)) {
+            // Create all directories required for the package
+            List.map(f -> {
+                let dir = Result.map(f, workingDir);
+                Result.flatMap(Path.mkdirs, dir)
+            }, dirs);
+
+            let quote = (s) -> Char.toString('"') + s + Char.toString('"');
+
+            // TODO: Abstract this
+            let newPackageFile = newFile([
+                "(package",
+                "  (name " + name + ")",
+                "  (version " + quote("0.1.0") + ")",
+                // Empty list/empty value?
+                "  (authors ((name " + quote("Rasmus Larsen") + ") (email " + quote("doctorherbstein@gmail.com") + ")))",
+                // Should description be quoted?
+                "  (description " + quote(name) + ")",
+             // "  (license GPLv3)",
+             // "  (license (name LGPLv2)",
+                "  (license (file " + quote("LICENSE") + ")",
+                "  (depends (other_package " + quote("1.2.3") + "))",
+                "  (home " + quote("https://github.com/Herbstein/package_one") + ")",
+                ")"
+            ]);
+            Result.flatMap(newPackageFile, Result.map(getPackageFile, workingDir));
```

Comment:
You may want to look into the `let*` syntax.

Reply:
Right now I'm trying to see if there's a nice way to use the `let*` syntax and still transform the error as previously described in relation to `mapErr` without hampering readability too much.